homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -35,22 +57,224 @@ class PassiveBluetoothEntityKey:
 
 _T = TypeVar("_T")
 
+STORAGE_KEY = "bluetooth.passive_update_processor"
+STORAGE_SAVE_INTERVAL = timedelta(minutes=15)
+PASSIVE_UPDATE_PROCESSOR = "passive_update_processor"
+
+
+@dataclasses.dataclass(slots=True, frozen=False)
+class PassiveBluetoothProcessorData:
+    """Data for the passive bluetooth processor."""
+
+    coordinators: set[PassiveBluetoothProcessorCoordinator] = dataclasses.field(
+        default_factory=set
+    )
+    all_restore_data: dict[
+        str, dict[str, RestoredPassiveBluetoothDataUpdate]
+    ] = dataclasses.field(default_factory=dict)
+
+
+class RestoredPassiveBluetoothDataUpdate(TypedDict):
+    """Restored PassiveBluetoothDataUpdate."""
+
+    devices: dict[str, DeviceInfo]
+    entity_descriptions: dict[str, dict[str, Any]]
+    entity_names: dict[str, str | None]
+    entity_data: dict[str, Any]
+
+
+def serialize_passive_bluetooth_entity_key(
+    entity_key: PassiveBluetoothEntityKey,
+) -> str:
+    """Serialize a passive bluetooth entity key."""
+    return f"{entity_key.key}___{entity_key.device_id or ''}"
+
+
+def deserialize_passive_bluetooth_entity_key(
```

Comment:
Let's find this as from_dict class method and as_dict to be consistent with other implementations